cmds/db-manager/main.go

```diff
@@ -0,0 +1,228 @@
+// Bootstrap script for Database deployment and migration
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"math"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/cockroachdb"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	dssCockroach "github.com/interuss/dss/pkg/cockroach"
+	"go.uber.org/zap"
+)
+
+// MyMigrate is an alias for exstending migrate.Migrate
+type MyMigrate struct {
+	*migrate.Migrate
+}
+
+// Direction is an alias for int indicating the direction and steps of migration
+type Direction int
+
+func (d Direction) String() string {
+	if d > 0 {
+		return "Up"
+	} else if d < 0 {
+		return "Down"
+	}
+	return "No Change"
+}
+
+var (
+	path      = flag.String("schemas_dir", "/Users/pcharlie/charlie_dss/dss/build/db-schemas/scd", "path to db migration files directory")
+	dbVersion = flag.String("db_version", "", "the db version to migrate to (ex: v1.0.0)")
+	step      = flag.Int("migration_step", 1, "the db migration step to go to")
+
+	cockroachParams = struct {
+		host    *string
+		port    *int
+		sslMode *string
+		sslDir  *string
+		user    *string
+	}{
+		host:    flag.String("cockroach_host", "0.0.0.0", "cockroach host to connect to"),
+		port:    flag.Int("cockroach_port", 26257, "cockroach port to connect to"),
+		sslMode: flag.String("cockroach_ssl_mode", "disable", "cockroach sslmode"),
+		user:    flag.String("cockroach_user", "root", "cockroach user to authenticate as"),
+		sslDir:  flag.String("cockroach_ssl_dir", "", "directory to ssl certificates. Must contain files: ca.crt, client.<user>.crt, client.<user>.key"),
+	}
+)
+
+func main() {
+	flag.Parse()
+	if *path == "" {
+		log.Panic("Must specify schemas_dir path")
+	}
+	if (*dbVersion == "" && *step == 0) || (*dbVersion != "" && *step != 0) {
+		log.Panic("Must specify a db_version or migration_step to goto")
+	}
+	uriParams := map[string]string{
+		"host":             *cockroachParams.host,
+		"port":             strconv.Itoa(*cockroachParams.port),
+		"user":             *cockroachParams.user,
+		"ssl_mode":         *cockroachParams.sslMode,
+		"ssl_dir":          *cockroachParams.sslDir,
+		"application_name": "SchemaManager",
+		"db_name":          filepath.Base(*path),
+	}
+	postgresURI, err := dssCockroach.BuildURI(uriParams)
+	if err != nil {
+		log.Panic("Failed to build URI", zap.Error(err))
+	}
+	crdbURI := strings.Replace(postgresURI, "postgresql", "cockroachdb", 1)
+	*path = fmt.Sprintf("file://%v", *path)
+	migrater, migraterErr := migrate.New(*path, crdbURI)
```

Comment:
In general, I think we [want to reuse the `err` name for errors](https://github.com/interuss/dss/pull/328#discussion_r442495313); no need for each error to be named differently.

Reply:
Since we are reassigning the migraterErr everytime we call .Steps() I think its best not to muddle the other err here so it could be used in the deferred .Close() call